src/effects/deeplearning/ModelManagerPanel.cpp

```diff
@@ -0,0 +1,809 @@
+                                       GetSize().GetWidth() - 30);
+   S.EndStatic();
+}
+
+void ModelCardPanel::PopulateMetadata(ShuttleGui &S)
+{
+   S.StartMultiColumn(2, wxALIGN_LEFT);
+   {
+      S.AddVariableText(XO("Effect: "))
+          ->SetFont(wxFont(wxFontInfo().Bold()));
+      S.AddVariableText(XO("%s")
+                            .Format(mCard->effect_type()));
+
+      S.AddVariableText(XO("Sample Rate: "))
+          ->SetFont(wxFont(wxFontInfo().Bold()));
+      S.AddVariableText(XO("%d")
+                            .Format(mCard->sample_rate()));
+
+      std::string tagString;
+      for (auto &tag : mCard->tags())
+      {
+         if (!tagString.empty())
+            tagString = tagString +  ", " + tag;
+         else
+            tagString = tag;
+      }
+      
+      S.AddVariableText(XO("Tags: "))
+          ->SetFont(wxFont(wxFontInfo().Bold()));
+      S.AddVariableText(XO("%s")
+                            .Format(tagString));
+   }
+   S.EndMultiColumn();
+}
+
+void ModelCardPanel::PopulateInstallCtrls(ShuttleGui &S)
+{
+   DeepModelManager &manager = DeepModelManager::Get();
+
+   S.StartVerticalLay(wxCENTER, true);
+   {
+      // add a progress gauge for downloads, but hide it
+      mInstallProgressGauge = safenew wxGauge(S.GetParent(), wxID_ANY, 100); // TODO:  sizing
+      mInstallProgressGauge->SetSize(wxSize(80, 20));
+      S.AddWindow(mInstallProgressGauge);
+      mInstallProgressGauge->Show();
+
+      bool installed = manager.IsInstalled(mCard);
+      TranslatableString status = installed ? XO("installed") : XO("uninstalled");
+      mInstallStatusText = S.AddVariableText(status, true);
+
+      InstallStatus iStatus = installed ? InstallStatus::installed : InstallStatus::uninstalled;
+      wxColour statusColor = mInstallStatusColors[iStatus];
+      mInstallStatusText->SetForegroundColour(statusColor);
+
+      // TODO: do translatable strings here from the begginign
+      TranslatableString cmd = installed ? XO("Uninstall") : XO("Install");
+      mInstallButton = S.AddButton(cmd);
+
+      SetInstallStatus(installed ? InstallStatus::installed : InstallStatus::uninstalled);
+
+      mSelectButton = S.AddButton(XO("Select"));
+      mSelectButton->Connect(wxEVT_COMMAND_BUTTON_CLICKED, 
+                        wxCommandEventHandler(ModelCardPanel::OnSelect), NULL, this);
+   }
+   S.EndVerticalLay();
+}
+
+void ModelCardPanel::PopulateMoreInfo(ShuttleGui &S)
+{
+   S.StartHorizontalLay(wxCENTER, true);
+   {
+      mMoreInfoButton = S.AddButton(XO("More Info"));
+      mMoreInfoButton->Connect(wxEVT_COMMAND_BUTTON_CLICKED, 
+                           wxCommandEventHandler(ModelCardPanel::OnMoreInfo), NULL, this);
+   }
+}
+
+void ModelCardPanel::FetchModelSize()
+{
+   DeepModelManager &manager = DeepModelManager::Get();
+
+   ModelSizeCallback onGetModelSize = [this](size_t size)
+   {
+      float sizeMB = (float)size / (float)(1024 * 1024);
```

Comment:
static_cast!

Reply:
resolved in https://github.com/hugofloresgarcia/audacity/commit/eb0ddcf13af9473788cc39c2b8f447b30894e955